homepage/src/App.js

```diff
@@ -3,24 +3,25 @@ import Home from './components/Home';
 import LibraryComponents from './components/LibraryComponents';
 import ClipLoader from 'react-spinners/ClipLoader';
 import {fetchContentfulAPI} from './imageRendering';
+import useImage from "./ImageContext";
 
 function App() {
   const [isHome, setIsHome] = useState(true);
   const [isLoading, setIsLoading] = useState(true);
-  const [imageURLS, setImageURLS] = useState([]);
+  const {imageURLS, setImageURLS} = useImage(); // this implicitly returns the imageURLS and setImageURLS state values/function 
+
 
-  // make the api call
   useEffect(() => {
     fetchContentfulAPI(imageURLS, setImageURLS, setIsLoading);
-  }, [imageURLS]);
+  }, []);
```

Comment:
So right here, I'm choosing to leave the array empty so that the request to contentful only gets executed only when the website first renders.

Reply:
<img width="845" alt="image" src="https://user-images.githubusercontent.com/54591248/154121587-de44495e-1898-4737-bb04-0070ad268c76.png">

Reply:
If you leave the array empty `useEffect` only runs on mount i.e. every time the object (in this case App.js) has a mount the function runs. However, we only want the useEffect hook to run ONCE on mount, and that is why we provide the dependency as this would prevent the variable from being set again unless it is changed. [This](https://stackoverflow.com/a/58579462) thread goes into a little more detail about this.